1-js/02-first-steps/12-nullish-coalescing-operator/article.md

```diff
@@ -1,93 +1,96 @@
-# Nullish coalescing operator '??'
+# Toán tử nullish coalescing '??'
 
 [recent browser="new"]
 
-Here, in this article, we'll say that an expression is "defined" when it's neither `null` nor `undefined`.
+Toán tử nullish coalescing được viết dưới dạng hai dấu hỏi `??`.
 
-The nullish coalescing operator is written as two question marks `??`.
+Do toán tử này coi `null` và `undefined` như nhau, chúng ta sẽ dùng một thuật ngữ đặc biệt ở bài này. Chúng ta sẽ nói rằng một biểu thức được coi là "được định nghĩa" khi nó không phải là `null` hoặc `undefined`.
 
-The result of `a ?? b` is:
-- if `a` is defined, then `a`,
-- if `a` isn't defined, then `b`.
+Kết quả của `a ?? b` là:
 
+- Nếu `a` được định nghĩa, trả về `a`,
+- Nếu `a` chưa được định nghĩa, trả về `b`.
 
-In other words, `??` returns the first argument if it's not `null/undefined`. Otherwise, the second one.
+Nói một cách khác, `??` trả về đối số đầu tiên nếu nó không có giá trị `null/undefined`. Nếu không thì trả về đối số thứ hai.
 
-The nullish coalescing operator isn't anything completely new. It's just a nice syntax to get the first "defined" value of the two.
+Toán tử nullish coalescing không phải là hoàn toàn mới. Nó đơn thuần là một cú pháp ngắn gọn để lấy giá trị đầu tiên "được định nghĩa" trong hai giá trị.
 
-We can rewrite `result = a ?? b` using the operators that we already know, like this:
+Chúng ta có viết lại biểu thức `result = a ?? b` sử dụng toán tử khác mà chúng ta đã biết, như sau:
 
 ```js
-result = (a !== null && a !== undefined) ? a : b;
+result = a !== null && a !== undefined ? a : b;
 ```
 
-The common use case for `??` is to provide a default value for a potentially undefined variable.
+Giờ thì chắc bạn đã biết rõ toán tử `??` làm gì. Hãy xem có thể áp dụng toán tử này ở đâu.
 
-For example, here we show `Anonymous` if `user` isn't defined:
+Use case thường gặp để dùng toán tử `??` là để cung cấp một giá trị mặc định cho một biến số có khả năng là undefined.
+
+Ví dụ, ở đây chúng ta sẽ hiển thị `user` nếu như được định nghĩa, nếu không sẽ hiển thị `Ẩn danh`:
 
 ```js run
 let user;
 
-alert(user ?? "Anonymous"); // Anonymous
+alert(user ?? "Ẩn danh"); // Ẩn danh (user chưa được định nghĩa)
 ```
 
-Of course, if `user` had any value except `null/undefined`, then we would see it instead:
+Còn dưới đây là ví dụ với `user` được gán với một cái tên:
 
 ```js run
 let user = "John";
 
-alert(user ?? "Anonymous"); // John
+alert(user ?? "Ẩn danh"); // John (user đã được định nghĩa)
 ```
 
-We can also use a sequence of `??` to select the first value from a list that isn't `null/undefined`.
+Chúng ta cũng có thể dùng nhiều toán tử `??` để chọn giá trị đầu tiên trong một danh sách mà không có giá trị là `null/undefined`.
```

Comment:
Sửa câu này lại như sau liệu có hợp lý hơn?

Chúng ta cũng có thể dùng nhiều toán tử `??` để chọn giá trị đầu tiên không phải là `null/undefined` trong một danh sách.